extensions/sql-bindings/src/common/azureFunctionsUtils.ts

```diff
@@ -594,3 +633,59 @@ export async function promptSelectDatabase(connectionInfo: IConnectionInfo): Pro
 	}
 	return selectedDatabase;
 }
+
+export async function getConnectionURI(connectionInfo: IConnectionInfo): Promise<string | undefined> {
+	const vscodeMssqlApi = await utils.getVscodeMssqlApi();
+
+	let connectionURI: string = '';
+	try {
+		await vscode.window.withProgress(
+			{
+				location: vscode.ProgressLocation.Notification,
+				title: constants.connectionProgressTitle,
+				cancellable: false
+			}, async (_progress, _token) => {
+				// show progress bar while connecting to the users selected connection profile
+				connectionURI = await vscodeMssqlApi.connect(connectionInfo!);
+			}
+		);
+	} catch (e) {
+		// mssql connection error will be shown to the user
+		return undefined;
+	}
+
+	return connectionURI;
+}
+
+export async function promptSelectTable(connectionURI: string, selectedDatabase: string, bindingType: BindingType): Promise<string | undefined> {
+	const vscodeMssqlApi = await utils.getVscodeMssqlApi();
+	const userObjectName = bindingType === BindingType.input ? constants.enterObjectName : constants.enterObjectNameToUpsert;
+
+	// Create query to get list of tables from database selected
+	let tableQuery = tablesQuery(selectedDatabase);
+	const params = { ownerUri: connectionURI, queryString: tableQuery };
+	let queryResult: azureFunctionsContracts.SimpleExecuteResult = await vscodeMssqlApi.sendRequest(azureFunctionsContracts.SimpleExecuteRequest.type, params);
+
+	// Get table names from query result rows
+	const tableNames = queryResult.rows.map(r => r[0].displayValue);
+	tableNames.push(userObjectName);
+	// prompt user to select table from list of tables
+	let selectedObject = await vscode.window.showQuickPick(tableNames, {
+		canPickMany: false,
+		title: constants.selectTable,
+		ignoreFocusOut: true
+	});
+
+	if (selectedObject === constants.enterObjectName) {
+		// user manually enters table or view to query or upsert into
+		return vscode.window.showInputBox({
+			prompt: bindingType === BindingType.input ? constants.sqlTableOrViewToQuery : constants.sqlTableToUpsert,
+			placeHolder: constants.placeHolderObject,
+			validateInput: input => input ? undefined : constants.nameMustNotBeEmpty,
+			ignoreFocusOut: true
+		}) ?? '';
+	}
+	return selectedObject;
+}
+
+export function tablesQuery(selectedDatabase: string): string { return `select concat('[',table_schema,']', '.', '[',table_name,']') from ${selectedDatabase}.INFORMATION_SCHEMA.TABLES where TABLE_TYPE = 'BASE TABLE'`; }
```

Comment:
Why do you qualify this with the DB anyways? Shouldn't we already be connecting to the DB to run this query? If you remove that then you can just make this another constant.

Reply:
Yeah, I will test it to make sure. Since we do set the connectionInfo.database prior the table promptSelectTable we would not need to state the selectedDatabase chosen.

Reply:
Tested this and if I remove selectedDatabase then the query would just result in getting the master database tables even if another is chosen. Not sure if this is due to connectionURI not having a reference to the database but I will keep this here in order to get the appropriate tables for each selected database.

Reply:
Please follow up with investigating this, it would be great not to have us making multiple different connections to the same server.

Reply:
Will make sure to make an issue to track this. We can make this experience better for sure.